src/CC.hs

```diff
@@ -0,0 +1,138 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+-- |
+--
+-- Utility for formatting @'Idea'@ data in accordance with the Code Climate
+-- spec: <https://github.com/codeclimate/spec>
+--
+module CC
+    ( printIssue
+    , fromIdea
+    ) where
+
+import Data.Aeson (ToJSON(..), (.=), encode, object)
+import Data.Char (toUpper)
+import Data.Monoid ((<>))
+import Data.Text (Text)
+import Language.Haskell.Exts.SrcLoc (SrcSpan(..))
+
+import qualified Data.Text as T
+import qualified Data.ByteString.Lazy.Char8 as C8
+
+import Idea (Idea(..), Severity(..))
+
+data Issue = Issue
+    { issueType :: Text
+    , issueCheckName :: Text
+    , issueDescription :: Text
+    , issueContent :: Text -- TODO: Markdown type?
+    , issueCategories :: [Text]
+    , issueLocation :: Location
+    , issueRemediationPoints :: Int
+    }
+
+data Location = Location FilePath Position Position
```

Comment:
I wonder if defining srcSpanJSON makes sense, so you avoid the orphan instance and avoid a separate location type. Would avoid multiple conversions and let you put SrcSpan in Issue?

Reply:
You're suggesting making a `ToJSON SrcSpan` and using it directly here? To me it feels wrong to make *the* JSON instance for anything conform CC's definition, so I'd lean towards a separate type anyway (even if just a `newtype` to allow a specific instance). Additionally, `SrcSpan -> Location` also handles relativizing the file-path (albeit naively at the moment). Taken together, these make the `Location` type wortwhile IMO.

Those are my reasons, but again it's totally your call if you'd like to do something else.